### ISSUE TITLE
Merge !993 break custom system command Newchatmembers. 

### ISSUE THREAD

**Issue by panaceya** [php-telegram-bot/core#994]
### 💥 BC Break Report
Merge !993 break custom system command Newchatmembers. 
Affected to all version started from 0.58.0 to 0.60.0

#### Required Information

|  ?                        |  !
| ---                       | ---
| PHP Telegram Bot version  | > 0.57.0
| PHP version               | 7.3.8
| Update Method             | Webhook
| Self-signed certificate   | no

#### Summary
this expression always false for Newchatmembers
https://github.com/php-telegram-bot/core/blob/06099b100195d49485a49cfeb56977248eb8f557/src/Telegram.php#L471

#### Temporary fix
- downgrade to 0.57.0
or
- add app/Commands/Newchatmembers.php (or another dir)
```php
protected $usage = '/newchatmembers';
```

**Comment by noplanman** [comment]
Hi @panaceya, this is due to the fact that system commands no longer get called directly and instead get handled by `GenericmessageCommand`. The problem was that it was possible to execute any system commands by simply calling that command directly in any chat, e.g. `/newchatmembers`.

This change has been flagged as a deprecation since 0.56.0 ([check changelog](https://github.com/php-telegram-bot/core/blob/master/CHANGELOG.md#0560---2019-04-15)).

You're absolutely right that this BC slipped in a bit too early without mentioning it directly 👍 

I'll correct that and thanks for letting me know!

Here is an example of how previous system commands would now be handled:
https://github.com/php-telegram-bot/support-bot/blob/4deebac7cd2525c0fdb7236150e05620b9747f95/commands/GenericmessageCommand.php#L47

**Comment by noplanman** [comment]
@jacklul Even though we were going towards `GenericmessageCommand` handling all non-user commands, I'm kind of rethinking this 🤔 

Does it maybe make sense to have a `ServicemessageCommand` of type `SystemCommand` that takes care of all non-user messages? It just seems weird to me now, that `GenericmessageCommand` should contain the code to be executed when a new user joins for example. "Generic message" should be just that, a user sent a "generic message" to the bot (be it text or any other type).

What do you think? I'm just trying to make this as user-friendly and logical as possible.

(Of course everyone is welcome to join the discussion 🎉)

**Comment by jacklul** [comment]
I honestly think that change was a good thing since all those extra service fields are still refering `Message` object. 

Maybe instead we could intruduce an extra code in `Genericmessage` that redirect some 'service' message to old system command names?

Something like this. maybe:
```php
    public function execute()
    {
        if ($deprecatedSystemCommandResult = $this->executeDeprecatedSystemCommand($this->getUpdate())) {
            return $deprecatedSystemCommandResult;
        }
    }

    public function executeDeprecatedSystemCommand(Update $update)
    {
        $message = $update->getMessage();

        if ($update->getEditedMessage()) {
            return $this->getTelegram()->executeCommand('editedmessage');
        }

        if ($update->getChannelPost()) {
            return $this->getTelegram()->executeCommand('channelpost');
        }

        if ($update->getEditedChannelPost()) {
            return $this->getTelegram()->executeCommand('editedchannelpost');
        }

        if ($message->getNewChatMembers()) {
            return $this->getTelegram()->executeCommand('newchatmembers');
        }

        if ($message->getLeftChatMember()) {
            return $this->getTelegram()->executeCommand('leftchatmember');
        }

        if ($message->getNewChatTitle()) {
            return $this->getTelegram()->executeCommand('newchattitle');
        }

        if ($message->getNewChatPhoto()) {
            return $this->getTelegram()->executeCommand('newchatphoto');
        }

        if ($message->getDeleteChatPhoto()) {
            return $this->getTelegram()->executeCommand('deletechatphoto');
        }

        if ($message->getGroupChatCreated()) {
            return $this->getTelegram()->executeCommand('groupchatcreated');
        }

        if ($message->getSupergroupChatCreated()) {
            return $this->getTelegram()->executeCommand('supergroupchatcreated');
        }

        if ($message->getChannelChatCreated()) {
            return $this->getTelegram()->executeCommand('channelchatcreated');
        }

        if ($message->getMigrateFromChatId()) {
            return $this->getTelegram()->executeCommand('migratefromchatid');
        }

        if ($message->getMigrateToChatId()) {
            return $this->getTelegram()->executeCommand('migratetochatid');
        }

        if ($message->getPinnedMessage()) {
            return $this->getTelegram()->executeCommand('pinnedmessage');
        }

        return null;
    }
```

In that case we should state in the readme that override `Genericmessage` will prevent those old commands from being executed.

**Comment by noplanman** [comment]
Just saw this crept in at version 0.58.0 with #970

I like your idea @jacklul. That will cause the deprecation notices to be called again though 🤔 
Since the obsolete system commands aren't called any more now anyway, I was thinking of just dropping them. I'll also build in a check, to make sure the custom command implementation exists, otherwise `GenericCommand` gets executed, which is obviously very not good.

I'll PR what I have and you can check it out.

*@noplanman reminds himself how this would all be over with a new command system... 🦄*

**Comment by noplanman** [comment]
Fixed with #999.
Old behaviour can be enforced by setting `SystemCommand::$execute_deprecated = true;` before handling the request.
